src/net.rs

```diff
@@ -91,23 +99,56 @@ pub enum MessageType {
     BlockGenerationRoundMessages(BlockGenerationRoundMessageType),
 }
 
+impl Display for MessageType {
+    fn fmt(&self, f: &mut Formatter<'_>) -> Result<(), Error> {
+        match self {
+            MessageType::KeyGenerationMessage(m) => write!(f, "{}", m),
+            MessageType::BlockGenerationRoundMessages(m) => write!(f, "{}", m),
```

Comment:
Preferably, the Message Type is for identifying an individual message. Should `KeyGenerationMessage` and `BlockGenerationRoundMessages` be defined as message types?

Reply:
I added it because It is useful to branch process by message type. But it going to be removed on federation management changes.